### PR TITLE
add opslevel_integration_custom_event_check resource

### PR DESCRIPTION
**Pull request by davidbloss**
Resolves [#158](https://github.com/OpsLevel/terraform-provider-opslevel/issues/158)

🎗️  relies on [terraform-opslevel-modules PR](https://github.com/OpsLevel/terraform-opslevel-modules/pull/38)

### Checklist

- [ ] I have run this code, and it appears to resolve the stated issue.
- [ ] This PR does not reduce total test coverage
- [ ] This PR has no user interface changes or has already received approval from product management to change the interface.
- [ ] Make a [changie](https://github.com/OpsLevel/terraform-provider-opslevel/blob/main/CONTRIBUTING.md#changie-change-log-generation) entry that explains the customer facing outcome of this change

### CODE REVIEW

File: opslevel/resource_opslevel_integration_custom_event_check.go [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+			},
+		},
+	}
+}
+
+func (r *IntegrationCustomEventCheckResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var planModel IntegrationCustomEventCheckResourceModel
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &planModel)...)
```
Comment (rocktavious) [inline opslevel/resource_opslevel_integration_custom_event_check.go:92]: would you mind converting these to use the `read[]` generic method?

File: tests/integration_custom_event_check.tftest.hcl [hunk 2]
```diff
@@ -0,0 +1,669 @@
...
+
+  assert {
+    condition = opslevel_integration_custom_event_check.this.type == var.type
+    error_message = format(
+      "expected '%v' but got '%v'",
+      var.type,
+      opslevel_integration_custom_event_check.this.type,
+    )
+  }
+}
```
Comment (rocktavious) [inline tests/integration_custom_event_check.tftest.hcl:110]: Should we add a CLI delete call in here somewhere to ensure it automatically re-creates them?

File: opslevel/resource_opslevel_integration_custom_event_check.go [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+		Type: RequiredStringValue(givenModel.Type.ValueString()),
+	}
+}
+
+func (r *IntegrationCustomEventCheckResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_integration_custom_event_check"
+}
+
+func (r *IntegrationCustomEventCheckResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	// validTypes := slices.Concat(opslevel.AllEventIntegrationEnum, []string{"filter", "framework", "language", "lifecycle", "owner", "product", "tag", "tier"})
```
Comment (rocktavious) [inline opslevel/resource_opslevel_integration_custom_event_check.go:53]: remove this?

File: opslevel/resource_opslevel_integration_custom_event_check.go [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+		return
+	}
+
+	input := opslevel.EventIntegrationUpdateInput{
+		Id:   opslevel.ID(planModel.Id.ValueString()),
+		Name: planModel.Name.ValueString(),
+	}
+
+	cecIntegration, err := r.client.UpdateEventIntegration(input)
+	if err != nil {
```
Comment (rocktavious) [inline opslevel/resource_opslevel_integration_custom_event_check.go:151]: Should we add the new opslevel api error stuff here if its not found?

**Review by rocktavious — CHANGES_REQUESTED** [review]
A few things to fix / improve

File: tests/integration_custom_event_check.tftest.hcl (hunk 2 above)
Comment (davidbloss) [reply tests/integration_custom_event_check.tftest.hcl]: Changing the type enacts a force replacement so we're good here. It's kind of a one off

**Review by rocktavious — APPROVED** [review]

**Pull request merged**